### PR TITLE
Fix issue #3975 - Added configurable IngressClass resource to helm chart

### PR DESCRIPTION
**Pull request by presidenten**
## Description
Added ingress class.

## Related Issues
Fixes issue #3975

## Testing
Installed emissary, verified with another helm chart with this template, installed with `--set myip=<node-public-ip>`:
```yaml
---

apiVersion: apps/v1
kind: Deployment
metadata:
  labels:
    app: hello
  name: hello
spec:
  replicas: 1
  selector:
    matchLabels:
      app: hello
  template:
    metadata:
      labels:
        app: hello
    spec:
      containers:
      - image: nginx
        name: nginx

---

apiVersion: v1
kind: Service
metadata:
  labels:
    app: hello
  name: hello
spec:
  type: ClusterIP
  selector:
    app: hello
  ports:
  - name: http
    port: 80
    protocol: TCP
    targetPort: 80

---

apiVersion: getambassador.io/v3alpha1
kind: Host
metadata:
  name: hello
spec:
  hostname: hello.{{ regexReplaceAll "\\." .Values.myip "-" }}.nip.io
  requestPolicy:
    insecure:
      action: Route

---

apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  name: hello
spec:
  ingressClassName: ambassador
  rules:
  - host: hello.{{ regexReplaceAll "\\." .Values.myip "-" }}.nip.io
    http:
      paths:
      - backend:
          service:
            name: hello
            port:
              number: 80
        path: /
        pathType: Prefix

```

## Checklist

Note! 
No changes to actual emissary ingress, only helm chart

 - [ ] I made sure to update `CHANGELOG.md`.
 - [X] This is unlikely to impact how Ambassador performs at scale.
 - [X] My change is adequately tested.
 - [X] I updated `DEVELOPING.md` with any any special dev tricks I had to use to work on this code efficiently.
   (No tricks needed)

### CODE REVIEW

**Comment by alexgervais** [comment]
Thanks for the contribution @presidenten ! While we review the PR, please take a moment to sign your commit to make the DCO check pass. See https://github.com/emissary-ingress/emissary/pull/4169/checks?check_run_id=5584651032 for details.

@AliceProxy, what do you think about adding a templated `IngressClass` by default? Knowing it's not supported by all Kubernetes versions, perhaps it would be safer to disable the automatic creation in `values.yaml` 🤔

**Comment by presidenten** [comment]
@alexgervais @AliceProxy 
We could have it disabled by default maybe?

**Review by AliceProxy — CHANGES_REQUESTED** [review]
Apologies for the very late review. Dont mind the failed checks, that's simply because your PR is from a fork and does not have access to the secrets in this repository for CI runs. I'll make a branch from these changes to get a CI run in.

This looks good to me although I will request an addition to the chart change log before we merge this.

File: charts/emissary-ingress/templates/controller-ingressclass.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{{- if .Values.controller.ingressClassResource.enabled -}}
```
Comment (AliceProxy) [inline charts/emissary-ingress/templates/controller-ingressclass.yaml:1]: There is no top level `controller` in your additions to the values. This should be just `.Values.ingressClassResource.enabled` to match your addition to the Values file.

Comment (presidenten) [reply charts/emissary-ingress/templates/controller-ingressclass.yaml:1]: > There is no top level `controller` in your additions to the values. This should be just `.Values.ingressClassResource.enabled` to match your addition to the Values file.

Nice find! Thanks!

I pulled latest from upstream master, fixed the `controller` thingie in values, and added info about the ingress class in changelog as well.

**Review by AliceProxy — APPROVED** [review]
Approved! Thanks so much for the contribution :)

**Pull request merged**